src/circuits/query.ts

```diff
@@ -213,64 +232,57 @@ async function parseRequest(
       fieldName: '',
     };
   }
+  if (Object.keys(query.credentialSubject).length > 1) {
+    throw new Error(`multiple requests not supported`);
+  }
 
-  let fieldName = '';
-  let fieldReq: Map<string, unknown>;
+  const txtSchema = new TextDecoder().decode(schema);
 
-  if (Object.keys(req).length > 1) {
-    throw new Error(`multiple requests  not supported`);
-  }
+  let fieldName: string;
+  let predicate: Map<string, unknown>;
 
-  for (const [key, value] of Object.entries(req)) {
+  for (const [key, value] of Object.entries(query.credentialSubject)) {
     fieldName = key;
 
-    fieldReq = value as Map<string, unknown>;
+    predicate = value as Map<string, unknown>;
 
-    if (Object.keys(fieldReq).length > 1) {
+    if (Object.keys(predicate).length > 1) {
       throw new Error(`multiple predicates for one field not supported`);
     }
     break;
   }
 
-  let operator: number;
-  const values: bigint[] = new Array<bigint>(valueLength).fill(BigInt(0));
-  for (const [key, value] of Object.entries(fieldReq)) {
-    if (!operators.has(key)) {
-      throw new Error(`operator is not supported by lib`);
-    }
-    operator = operators.get(key);
-
-    if (Array.isArray(value)) {
-      for (let index = 0; index < value.length; index++) {
-        values[index] = BigInt(value[index]);
-      }
-    } else {
-      values[0] = BigInt(value as string);
-    }
-    break;
+  let datatype: string;
+  if (fieldName !== '') {
+    datatype = await Path.newTypeFromContext(
+      txtSchema,
+      `${query.type}.${fieldName}`,
+    );
   }
 
-  let slotIndex: number;
-  let claimPathKey: bigint;
-  if (merkalized === 1) {
-    const txtSchema = new TextDecoder().decode(schema);
-    const path = await Path.getContextPathKey(txtSchema, credType, fieldName);
-    path.prepend(['https://www.w3.org/2018/credentials#credentialSubject']);
-    claimPathKey = await path.mtEntry();
-  } else {
-    slotIndex = getFieldSlotIndex(fieldName, credType, schema);
-  }
+  const [operator, values] = await parsePredicate(predicate, datatype);
+  const zeros: Array<bigint> = Array.from({
+    length: outputs.valueArraySize - values.length,
+  }).fill(BigInt(0)) as Array<bigint>;
```

Comment:
I guess you can delete `as Array<bigint>` here

Reply:
I have the error: `Type 'unknown[]' is not assignable to type 'bigint[]'`